golem_messages/validators.py

```diff
@@ -19,3 +19,103 @@ def validate_integer(field_name, value):
             field=field_name,
             value=value,
         )
+
+
+def is_number_positive(value, value_type):
+    return isinstance(value, value_type) and value > 0
+
+
+def validate_positive_integer_value(field_name, value):
+    if not is_number_positive(value, int):
+        raise exceptions.FieldError(
+            "should be a positive integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_not_negative_integer_value(field_name, value):
+    if not isinstance(value, int) or value < 0:
+        raise exceptions.FieldError(
+            "should be a not negative integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_boolean(field_name, value):
+    if not isinstance(value, bool):
+        raise exceptions.FieldError(
+            f"{field_name} should be boolean",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list(field_name, value, min_length, max_length):
+    if not (isinstance(value, list) and
+            (min_length <= len(value) <= max_length)):
+        raise exceptions.FieldError(
+            f"{field_name} should be a list with minimum {min_length}"
+            f" element and maximum {max_length} elements.",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list_of_positive_integers(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not is_number_positive(element, int):
+            raise exceptions.FieldError(
+                f"{field_name} should be a list with positive integers",
+                field=field_name,
+                value=value
+            )
+
+
+def validate_list_of_non_negative_floats(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not is_number_positive(element, float) and float(element) != 0.0:
+            raise exceptions.FieldError(
+                f"{field_name} should be a list of floats equal or greater 0.0",
+                field=field_name,
+                value=element
+            )
+
+
+def validate_enum(field_name, value, enum_class):
+    try:
+        enum_class(value)
+    except ValueError:
+        raise exceptions.FieldError(
+            f"{field_name} has not valid value for"
+            f" {enum_class.__name__} enum class",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_task_type_with_meta_parameters(field_name, value, valid_pairs):
```

Comment:
@kbeker docstring please -> and generally, it would be good to have docstrings for each function in general ... they can be optional if the name of the method unambiguously and perspicuously describes the purpose... otherwise, it's good for everyone reading the code to have a clear explanation, especially in case of reusable stuff like validators...